File: CSharp/Day5.cs
using System;
using System.IO;
using System.Collections.Generic;

namespace CSharp
{
    class Day5
    {
        public static void getSolutions()
        {
            string[] inputData = File.ReadAllLines("../../../Day5Input.txt");
            RunPart1(inputData);
        }

        static void RunPart1(string[] inputData)
        {
            List<int> seatNumbers = new();
            foreach (string line in inputData)
            {
                seatNumbers.Add(Convert.ToInt32(line.Replace("F", "0").Replace("B", "1").Replace("L", "0").Replace("R", "1"), 2));
            }
            seatNumbers.Sort();
            seatNumbers.Reverse();
            Console.WriteLine(seatNumbers[0]);
            RunPart2(seatNumbers);
        }

        static void RunPart2(List<int> seatNumbers)
        {
            for (int i = 1; i < seatNumbers.Count - 1; i++)
            {
                if (seatNumbers[i] - seatNumbers[i + 1] > 1)
                {
                    Console.WriteLine(seatNumbers[i] - 1);
                    return;
                }
            }
        }
    }
}

File: Python/Day5.py
inputString = open("Python/Day5Input.txt", "r").read().splitlines()
seatList = []

def readSeats():
    for scan in inputString:
        seatList.append(int(scan.replace("F", "0").replace("B", "1").replace("R", "1").replace("L", "0"), 2))
    seatList.sort(reverse=True)

def findSeat():
    for seat in seatList:
        if seat - 2 in seatList and seat - 1 not in seatList: return seat - 1

readSeats()
print(seatList[0])
print(findSeat())
